src/coreclr/pal/src/map/virtual.cpp

```diff
@@ -663,6 +667,9 @@ VIRTUALCommitMemory(
 
     if ( !pInformation )
     {
+        /* Set if VIRTUALReserveMemory will allocate new memory from the kernel. */
+        IsNewMemory = (((flAllocationType & MEM_RESERVE_EXECUTABLE) != 0) && (lpAddress == NULL));
```

Comment:
I don't think this is correct. The `VIRTUALReserveMemory` can get new memory from the kernel even when the `MEM_RESERVE_EXECUTABLE` is set in case we deplete the initially reserved executable memory. Also, the initially reserved memory is marked as `MADV_DODUMP`, so we need to set the advise to `MADV_DUMP` for it.